### PR TITLE
Create action to automate deploy

### PR DESCRIPTION
**Pull request by mtullyoc**
[Linear BAL-3](https://linear.app/bleu-llc/issue/BAL-37/create-action-to-automate-deployment-on-push-to-main)

### CODE REVIEW

**Comment by mtullyoc** [comment]
@ribeirojose 👀

File: .github/workflows/graph.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          pnpm codegen
+          cp subgraph.goerli.yaml subgraph.yaml
+          pnpm build
+
+      - uses: gtaschuk/graph-deploy@v0.1.12
+        with:
+          graph_access_token: ${{secrets.GRAPH_ACCESS_TOKEN}}
+          graph_subgraph_name: "balancer-pools-metadata"
+          graph_account: "bleu-llc"
+          graph_config_file: "./packages/subgraph/subgraph.yaml"
```
Comment (mtullyoc) [inline .github/workflows/graph.yml:33]: Should we use `working-directory: ./packages/subgraph` here?

File: .github/workflows/graph.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Deploy Graph
+
+on:
+  push:
+    branches: master
```
Comment (ribeirojose) [inline .github/workflows/graph.yml:5]:
```suggestion
    branches: main
```
would also add workflow-dispatch to make it easier for us to test it in this PR

File: .github/workflows/graph.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+jobs:
+  deploy-goerli:
+    runs-on: ubuntu-latest
+    environment: graph
+    steps:
+      - uses: actions/checkout@v3
+
```
Comment (ribeirojose) [inline .github/workflows/graph.yml:13]: depending on what we do we'd need to recur submodules here

File: .github/workflows/graph.yml (hunk 1 above)
Comment (ribeirojose) [reply .github/workflows/graph.yml:33]: Idk, does that work? 😬

File: packages/subgraph/abis/PoolMetadataRegistry.json [hunk 4]
```diff
@@ -1,54 +1,1958 @@
-[
-  {
-    "inputs": [
-      { "internalType": "contract IVault", "name": "vault", "type": "address" }
-    ],
-    "stateMutability": "nonpayable",
-    "type": "constructor"
+{
+  "abi": [
```
Comment (ribeirojose) [inline packages/subgraph/abis/PoolMetadataRegistry.json:2]: I'm not sure this is what you want. I think you probably want just what's nested under this `abi` key for this subgraph

**Review by ribeirojose — COMMENTED** [review]
have you tested this? I'm not comfortable giving an approval without seeing a successful workflow run and the deployed subgraph

File: .github/workflows/graph.yml (hunk 1 above)
Comment (mtullyoc) [reply .github/workflows/graph.yml:36]: almost 😬... let's fix the block number

File: .github/workflows/graph.yml (hunk 3 above)
Comment (mtullyoc) [reply .github/workflows/graph.yml:16]: I think would be necessary if we needed to run `forge` on contracts package... here we worked on subgraph folder only ✌🏻

File: .github/workflows/graph.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+name: Deploy Graph
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - tullyo/bal-37-create-action-to-automate-deployment-on
```
Comment (mendesfabio) [inline .github/workflows/graph.yml:8]:
```suggestion
```

**Review by mendesfabio — APPROVED** [review]
lgtm!

**Review by mendesfabio — APPROVED** [review]

**Pull request merged**